sources/packages/api/src/route-controllers/institution/institution.controller.ts

```diff
@@ -60,4 +61,31 @@ export class InstitutionController extends BaseController {
   async sync(@UserToken() token: IUserToken) {
     await this.institutionService.syncInstitution(token);
   }
+
+  @Get("/users")
+  async allUsers(
+    @UserToken() user: IUserToken,
+  ): Promise<InstitutionUserRespDto[]> {
+    const institution = await this.institutionService.getInstituteByUserName(
+      user.userName,
+    );
+    return (await this.institutionService.allUsers(institution.id)).map(
```

Comment:
IMHO there are too many things happening inside this return. Is it possible to break the logic into small pieces to make it easier to read it?